cpmpy/tools/explain/mus.py

```diff
@@ -108,6 +111,55 @@ def do_recursion(soft, hard, delta):
     return [dmap[a] for a in core]
 
 
+def optimal_mus(soft, hard=[], weights=1, solver="ortools", hs_solver="ortools"):
+    """
+        Find an optimal MUS according to a linear objective function.
+        Works by iteratively generating correction subsets and computing optimal hitting sets to those enumerated sets.
+        For better performance of the algorithm, use an incemental solver to compute the hitting sets such as Gurobi.
+
+        Assumption-based implementation for solvers that support s.solve(assumptions=...)
+        More naive version available as `optimal_mus_naive` to use with other solvers.
+
+        CPMpy implementation loosely based on the "SMUS" algorithm from
+            Ignatiev, Alexey, et al. "Smallest MUS extraction with minimal hitting set dualization."
+            International Conference on Principles and Practice of Constraint Programming. Cham: Springer International Publishing, 2015.
+        and OCUS from:
+            Gamba, Emilio, Bart Bogaerts, and Tias Guns. "Efficiently explaining CSPs with unsatisfiable subset optimization."
+            Journal of Artificial Intelligence Research 78 (2023): 709-746.
+
+    """
+
+
+    model, soft, assump = make_assump_model(soft, hard)
+    dmap = dict(zip(assump, soft))
```

Comment:
maybe 'cmap'? with comment  # map of assumption variable to constraint

Reply:
In the other's it's also dmap... Not sure why tho ;)